src/table/Exposure.cc

```diff
@@ -121,15 +122,23 @@ struct PersistenceHelper {
     // Read psf, wcs, etc. from an archive to an ExposureRecord
     void readRecord(
         BaseRecord const & input, ExposureRecord & output,
-        SchemaMapper const & mapper, io::InputArchive const & archive
+        SchemaMapper const & mapper, io::InputArchive const & archive,
+        int tableVersion
     ) const {
         output.assign(input, mapper);
         output.setPsf(archive.get<detection::Psf>(input.get(psf)));
         output.setWcs(archive.get<image::Wcs>(input.get(wcs)));
         output.setCalib(archive.get<image::Calib>(input.get(calib)));
         output.setApCorrMap(archive.get<image::ApCorrMap>(input.get(apCorrMap)));
         output.setValidPolygon(archive.get<geom::polygon::Polygon>(input.get(validPolygon)));
-        output.setVisitInfo(archive.get<image::VisitInfo>(input.get(visitInfo)));
+        if (tableVersion > 1) {
+            output.setVisitInfo(archive.get<image::VisitInfo>(input.get(visitInfo)));
+        } else {
+            // VisitInfo not available; set to invalid values
+            daf::base::PropertySet md;
+            auto nullVisitInfoPtr = std::make_shared<image::VisitInfo>(md);
+            output.setVisitInfo(nullVisitInfoPtr);
+        }
```

Comment:
I'm a little surprised you consider it better to default-construct a `VisitInfo` here rather than leave it an empty pointer.  Is having a non-null `VisitInfo` required for all `Exposure` objects (even coadd?!) and hence something you want to enforce for `ExposureRecords` as well?